services/prototype.go

```diff
@@ -475,10 +471,15 @@ func searchDatabase(_ context.Context,
 		}
 	}
 
-	// FIXME: for now, if a user ORCID is not specified, use the client's ORCID
+	// FIXME: for now, if a user ORCID is not specified, use the user/client's ORCID
 	orcid := input.Orcid
 	if orcid == "" {
-		orcid = client.Orcid
+		switch u := userOrClient.(type) {
+		case auth.User:
+			orcid = u.Orcid
+		case auth.Client:
+			orcid = u.Orcid
```

Comment:
Is it not possible to combine these cases? Are other cases anticipated to make the `switch` a better choice than using `if` statements? (I know, this is a small and very petty point.)